packages/mobile/src/Drawers.tsx

```diff
@@ -19,32 +23,91 @@ import { TransferAudioMobileDrawer } from 'app/components/transfer-audio-mobile-
 import { TrendingFilterDrawer } from 'app/screens/trending-screen'
 
+import { useDrawerState } from './components/drawer'
+import { useNativeDrawer } from './hooks/useDrawer'
+import { Drawer } from './store/drawers/slice'
+
+type CommonDrawerProps = {
+  modal: ComponentType
+  modalName: Modals
+}
+
+const CommonDrawer = ({ modal: Modal, modalName }: CommonDrawerProps) => {
+  const { modalState } = useDrawerState(modalName)
+
+  if (modalState === false) return null
+
+  return <Modal />
+}
+
+type NativeDrawerProps = {
+  drawer: ComponentType
+  drawerName: Drawer
+}
+
+const NativeDrawer = ({ drawer: Drawer, drawerName }: NativeDrawerProps) => {
+  const { visibleState } = useNativeDrawer(drawerName)
+
+  if (visibleState === false) return null
+
+  return <Drawer />
+}
+
+const commonDrawersMap: { [Modal in Modals]?: ComponentType } = {
+  TiersExplainer: TiersExplainerDrawer,
+  TrendingRewardsExplainer: TrendingRewardsDrawer,
+  ChallengeRewardsExplainer: ChallengeRewardsDrawer,
+  APIRewardsExplainer: ApiRewardsDrawer,
+  TransferAudioMobileWarning: TransferAudioMobileDrawer,
+  MobileConnectWalletsDrawer: ConnectWalletsDrawer,
+  MobileEditCollectiblesDrawer: EditCollectiblesDrawer,
+  Share: ShareDrawer,
+  ShareSoundToTikTok: ShareToTikTokDrawer,
+  CollectibleDetails: CollectibleDetailsDrawer,
+  DeactivateAccountConfirmation: DeactivateAccountConfirmationDrawer,
+  Cognito: CognitoDrawer,
+  FeedFilter: FeedFilterDrawer,
+  TrendingGenreSelection: TrendingFilterDrawer,
+  MobileUpload: MobileUploadDrawer,
+  Overflow: OverflowMenuDrawer
+  /* Disable the audio breakdown drawer until we get
+   * the feature flags to work for native mobile
+   * AudioBreakdown: AudioBreakdownDrawer
+   */
+}
+
+const nativeDrawersMap: { [DrawerName in Drawer]?: ComponentType } = {
+  EnablePushNotifications: EnablePushNotificationsDrawer,
+  DownloadTrackProgress: DownloadTrackProgressDrawer,
+  ForgotPassword: ForgotPasswordDrawer
+}
+
+const commonDrawers = Object.entries(commonDrawersMap) as [
+  Modals,
+  ComponentType
+][]
+
+const nativeDrawers = Object.entries(nativeDrawersMap) as [
+  Drawer,
+  ComponentType
+][]
+
 export const Drawers = () => {
   return (
     <>
-      <MobileUploadDrawer />
-      <EnablePushNotificationsDrawer />
-      <CollectibleDetailsDrawer />
-      <ConnectWalletsDrawer />
-      <EditCollectiblesDrawer />
-      <OverflowMenuDrawer />
-      <DeactivateAccountConfirmationDrawer />
-      <DownloadTrackProgressDrawer />
-      <TransferAudioMobileDrawer />
-      <TrendingRewardsDrawer />
-      <ApiRewardsDrawer />
+      {commonDrawers.map(([modalName, Modal]) => {
+        return (
+          <CommonDrawer modal={Modal} modalName={modalName} key={modalName} />
+        )
+      })}
+      {nativeDrawers.map(([drawerName, Drawer]) => (
+        <NativeDrawer
+          key={drawerName}
+          drawerName={drawerName}
+          drawer={Drawer}
+        />
+      ))}
       <AddToPlaylistDrawer />
```

Comment:
why is add to playlist separated?

Reply:
so the drawers that use their own slice to manage open/closed cant use the `useDrawer` hooks etc. im down to try and migrate this one though